cmd/app/main.go

```diff
@@ -60,11 +64,38 @@ func main() {
 		log.Fatal(err)
 	}
 
-	server := virtual.NewServer(reg, environment)
+	var server Server = virtual.NewServer(reg, environment)
```

Comment:
The other alternative to specifying the interface is `server := Server(virtual.NewServer(reg, environment))`, which needs to be more readable. And the interface is needed since `virtual.server` is private